### DISCUSSION TITLE
Guidance on spring settings?

### DISCUSSION THREAD

**Discussion by DennisSmolek** [jrouwe/JoltPhysics#1007] [Q&A]
I'm working on a physically based rig and having the hardest time understanding/managing springs.
I have a Distance constraint that has a second body (light blue ball) following the first (capsule in character controller) and coming to rest at the first body's position. 
Chasing:
![Screenshot 2024-03-26 230504](https://github.com/jrouwe/JoltPhysics/assets/1397052/65779d4b-b367-4d24-8a3d-cf43420cf815)

At Rest:
![Screenshot 2024-03-26 230517](https://github.com/jrouwe/JoltPhysics/assets/1397052/11f12d5e-7811-4d07-980f-f722e1570c80)

I'm trying to tweak these delays/strengths of following. 

Right now with the spring disabled it works well enough but the moment I try and tweak the spring settings it become unmanageable. 

With frequency mode it's either too slow or when it's fast enough it overshoots over and over. Damping just slows down the spring simulation. 

With Stiffness mode I can't really figure out what the numbers are doing, but again, when it's fast enough (like when its disabled) it never comes to a rest and bounces forever...

Does anyone have insights on tuning the springs?

**Comment by jrouwe** [comment]
Did you read the [documentation](https://jrouwe.github.io/JoltPhysics/#constraint-motors)?

**Comment by DennisSmolek** [comment]
🤦
The heading of that section was for motors which I haven't messed with much yet so didn't read. 
I've been trying to tweak the soft constraints...

I'll give those standards a shot and see what happens!

**Comment by DennisSmolek** [comment]
Ok so even with Frequency at 20 and the damping at 1 like the docs recommend for a stiff spring the second body takes forever to come to  a rest...
I wonder if the object is colliding or something

**Comment by jrouwe** [comment]
You closed the discussion so I'm guessing it's resolved?

In any case, damping = 1 is critical damping, which is the damping where there should not be any oscillation. The reason I pointed to the documentation for motors is that springs on motors work in the exact same way as springs on a DistanceConstraint.

**Reply by DennisSmolek** [reply]
I closed it because I should have done a better job reading the docs lol.

my object still oscillates, however I’m thinking it’s not properly filtering collisions so what is happening is the collision and constraint system are at odds with each other. That’s my best guess at the moment. I’ve written my own spring/rubberbanding implementation for now, but I’ll readdress it when i start working on motors
